FeedSX/screens/UniversalFeed/index.tsx

```diff
@@ -93,6 +112,90 @@ const UniversalFeed = () => {
     return getFeedResponse;
   }
 
+  // this useEffect handles the execution of addPost api
+  useEffect(() => {
+    if (
+      mediaAttachmemnts.length > 0 ||
+      linkAttachments.length > 0 ||
+      postContent != ''
+    ) {
+      setPostUploading(true);
+      postAdd();
+    }
+  }, [mediaAttachmemnts, linkAttachments, postContent]);
+
+  // aws configuration
+  function getAWS() {
+    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+    (AWS.config.region = process.env.AWS_REGION),
+      (AWS.config.credentials = new AWS.CognitoIdentityCredentials({
+        IdentityPoolId: `${process.env.AWS_IDENTITY_POOL_ID}`,
+      }));
+    const s3 = new AWS.S3({
+      apiVersion: process.env.S3_API_VERSION,
+      params: {Bucket: process.env.S3_BUCKET},
+    });
+    return s3;
+  }
+
+  // this function uploads the media on AWS S3 bucket
+  const uploadMedia = async (media: any, userUniqueId: string) => {
+    const response = await fetch(media.url);
+    const blob = await response.blob();
+    let mediaObject = getAWS()
+      .upload({
+        Key: `files/post/${userUniqueId}/${media.name}`,
+        Bucket: `${process.env.S3_BUCKET}`,
+        Body: blob,
+        ACL: `public-read-write`,
+        ContentType: media.format,
+      })
+      .on('httpUploadProgress', function (progress) {
+        Math.round((progress.loaded / progress.total) * 100);
+      });
+    return mediaObject.promise();
+  };
```

Comment:
can we move this whole upload related code to a util file as `UploadUtils` and use the code from there itself.

Reply:
resolved